sopel/modules/help.py

```diff
@@ -14,17 +14,54 @@
 import collections
 import requests
 
+from sopel.config.types import ValidatedAttribute, StaticSection
 from sopel.logger import get_logger
 from sopel.module import commands, rule, example, priority
 
 logger = get_logger(__name__)
 
 
+class HelpSection(StaticSection):
+    show_server_host = ValidatedAttribute('show_server_host', bool, default=True)
+    """Show the server's hostname/ip in the first line of the help listing?"""
+
+
+def configure(config):
+    config.define_section('help', HelpSection)
+    config.help.configure_setting(
+        'show_server_host',
+        'Should the help command show the server hostname/IP in the listing?'
+    )
+
+
 def setup(bot):
-    global help_prefix
+    global help_prefix, help_track_settings
+
+    bot.config.define_section('help', HelpSection)
+
+    # Put name of settings that should be tracked in cache validation in
+    # 'help-track-settings' list below; i.e., those that will require the help
+    # listing to be regenerated, or re-POSTed to paste.
+    help_track_settings = [
+        'show_server_host'
+    ]
+    update_cache(bot)  # Initialize cache with first call
+
     help_prefix = bot.config.core.help_prefix
 
 
+def update_cache(bot):
+    for setting in help_track_settings:
+        bot.memory['help-cache-{}'.format(setting)] = getattr(bot.config.help, setting)
+
+
+def is_cache_valid(bot):
+    for setting in help_track_settings:
+        if bot.memory['help-cache-{}'.format(setting)] != getattr(bot.config.help, setting):
```

Comment:
Of course, the elegance of this line would be somewhat lost if you take my suggestion above about the memory structure. :/ (This whole function would have to be rewritten, in fact.)

Reply:
Done. Just had to nest `getattr`s :open_mouth: